lib/paths-cache.js

```diff
@@ -20,6 +20,19 @@ export default class PathsCache extends EventEmitter {
     this._filePathsByProjectDirectory = new Map()
     this._filePathsByDirectory = new Map()
     this._fileWatchersByDirectory = new Map()
+
+    updateConfig()
+  }
+
+  updateConfig() {
+    this.config = {
+      excludeVcsIgnoredPaths: atom.config.get('core.excludeVcsIgnoredPaths'),
+      ignoreSubmodules: atom.config.get('autocomplete-paths.ignoreSubmodules'),
+      shouldIgnoredNames: atom.config.get('autocomplete-paths.ignoredNames'),
+      ignoredNames: atom.config.get('core.ignoredNames'),
+      ignoredPatterns: atom.config.get('autocomplete-paths.ignoredPatterns'),
+      maxFileCount: atom.config.get('autocomplete-paths.maxFileCount')
```

Comment:
`maxFileCount` is not observed and updated when changed.

Reply:
With these changes you should also write tests to prevent any regressions.

Reply:
Fixed in https://github.com/atom-community/autocomplete-paths/pull/237/commits/ebbfea12d98dfa31dd166a36e909d95ac7870d41

Reply:
> With these changes you should also write tests to prevent any regressions.

I did not add any features, or I did not change any functionality. This is just some refactoring. It does not need new tests.

Reply:
It should already have tests to make sure changing settings updates the functionality. Since you are affecting that part of the code you should add those tests even if it is just refactoring.